crates/cheatcodes/src/inspector.rs

```diff
@@ -1862,6 +1905,7 @@ where {
             accesses: self.accesses.as_mut(),
             persisted_factory_deps: Some(&mut self.persisted_factory_deps),
             paymaster_data: self.paymaster_params.take(),
+            persist_nonce_update: self.broadcast.is_some() || self.zk_persist_nonce_update.get(),
```

Comment:
We need to avoid a similar short-circuit bug as we fixed above:

```rs
let zk_persist_nonce_update = self.zk_persist_nonce_update.get();
..
persist_nonce_update: self.broadcast.is_some() || zk_persist_nonce_update,
```